### PR TITLE
Add FiscalDay

### PR DESCRIPTION
**Pull request by nicmendoza**
Hey @adamjstewart here's a possible implementation. Let me know what you think. I can add documentation if/when you feel good about the PR.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Since there is overlap between this and your FiscalMonth PR, I'm going to focus on reviewing the other PR first.

**Comment by nicmendoza** [comment]
Merged master, so should be good to review now.

File: fiscalyear.py [hunk 1]
```diff
@@ -815,6 +864,220 @@ def __ge__(self, other):
...
+
+    __slots__ = ['_fiscal_year', '_fiscal_day']
+
+    def __new__(cls, fiscal_year, fiscal_day):
+        """Constructor.
+
+        :param fiscal_year: The fiscal year
+        :type fiscal_year: int or str
+        :param fiscal_year: The fiscal year
+        :type fiscal_year: int or str
```
Comment (adamjstewart) [inline fiscalyear.py:878]:
```suggestion
        :param fiscal_day: The fiscal day
        :type fiscal_day: int or str
```

File: fiscalyear.py [hunk 2]
```diff
@@ -815,6 +864,220 @@ def __ge__(self, other):
...
+        """
+        today = FiscalDate.today()
+        return cls(today.fiscal_year, today.fiscal_day)
+
+    def __repr__(self):
+        """Convert to formal string, for repr().
+
+        >>> fd = FiscalDay(2017, 1)
+        >>> repr(fd)
+        'FiscalDay(2017,1)'
```
Comment (adamjstewart) [inline fiscalyear.py:908]:
```suggestion
        'FiscalDay(2017, 1)'
```

File: fiscalyear.py [hunk 3]
```diff
@@ -815,6 +864,220 @@ def __ge__(self, other):
...
+        return 'FY%d FD%d' % (self._fiscal_year, self._fiscal_day)
+
+    # TODO: Implement __format__ so that you can print
+    # fiscal year as 17 or 2017 (%y or %Y)
+
+    def __contains__(self, item):
+        """Returns True if item in self, else False.
+
+        :param item: The item to check
+        :type item: FiscalYear, FiscalQuarter, FiscalDateTime,
```
Comment (adamjstewart) [inline fiscalyear.py:930]:
```suggestion
        :type item: FiscalDay, FiscalDateTime,
```

File: fiscalyear.py [hunk 4]
```diff
@@ -815,6 +864,220 @@ def __ge__(self, other):
...
+        return self._fiscal_day
+
+    @property
+    def start(self):
+        """:returns: Start of the fiscal day
+        :rtype: FiscalDateTime
+        """
+
+        fiscal_year = FiscalYear(self._fiscal_year)
+        days_elapsed = datetime.timedelta(days=self._fiscal_day - 1)
```
Comment (adamjstewart) [inline fiscalyear.py:981]: Does this work when `self._fiscal_day == 1`?

File: fiscalyear.py [hunk 5]
```diff
@@ -815,6 +864,220 @@ def __ge__(self, other):
...
+        """:returns: The previous fiscal day
+        :rtype: FiscalDay
+        """
+        fiscal_year = self._fiscal_year
+        fiscal_day = self._fiscal_day - 1
+        if fiscal_day == 0:
+            fiscal_year -= 1
+            try:
+                fiscal_day = _check_fiscal_day(fiscal_year, 366)
+            except Exception:
```
Comment (adamjstewart) [inline fiscalyear.py:1011]:
```suggestion
            except ValueError:
```

File: fiscalyear.py [hunk 6]
```diff
@@ -815,6 +864,220 @@ def __ge__(self, other):
...
+
+    @property
+    def next_fiscal_day(self):
+        """:returns: The next fiscal day
+        :rtype: FiscalDay
+        """
+        fiscal_year = self._fiscal_year
+        try:
+            fiscal_day = _check_fiscal_day(fiscal_year, self._fiscal_day + 1)
+        except Exception:
```
Comment (adamjstewart) [inline fiscalyear.py:1024]:
```suggestion
        except ValueError:
```

File: fiscalyear.py [hunk 7]
```diff
@@ -842,6 +1105,19 @@ def fiscal_month(self):
...
+    def fiscal_day(self):
+        """:returns: The fiscal day
+        :rtype: int
+        """
+        day_of_calendar_year = self.timetuple().tm_yday
+        fiscal_year = FiscalYear(self.fiscal_year)
+        beginning_of_fiscal_year = fiscal_year.start.timetuple().tm_yday
+        days_elapsed = day_of_calendar_year - beginning_of_fiscal_year + 1
+        if days_elapsed < 1:
+            days_elapsed = (365 if fiscal_year.isleap else 366) + days_elapsed
```
Comment (adamjstewart) [inline fiscalyear.py:1118]:
```suggestion
            days_elapsed += (365 if fiscal_year.isleap else 366)
```

File: test_fiscalyear.py [hunk 8]
```diff
@@ -395,6 +403,23 @@ def test_q3(self, a):
...
+            assert a.isleap is False
+            assert b.isleap is True
+
+        with fiscalyear.fiscal_calendar(start_year='same', start_month=3):
+            assert a.isleap is False
+            assert g.isleap is True
+
+        with fiscalyear.fiscal_calendar(start_year='same', start_month=1):
+            assert a.isleap is True
+            assert g.isleap is False
```
Comment (adamjstewart) [inline test_fiscalyear.py:421]: I'm always hesitant when I see `is` used in Python. Personally, I only use `is` to compare against `None`. I would instead do:
```suggestion
    def test_is_leap(self, a, b, g):
        # default US start_year='previous', start_month=10
        assert isinstance(a.isleap, bool)
        assert isinstance(g.isleap, bool)

        with fiscalyear.fiscal_calendar(start_year='previous', start_month=1):
            assert not a.isleap
            assert b.isleap

        with fiscalyear.fiscal_calendar(start_year='same', start_month=3):
            assert not a.isleap
            assert g.isleap

        with fiscalyear.fiscal_calendar(start_year='same', start_month=1):
            assert a.isleap
            assert not g.isleap
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: fiscalyear.py (hunk 4 above)
Comment (nicmendoza) [reply fiscalyear.py:977]: Zero days have elapsed on fiscal day one so it's year start plus zero on `self.fiscal_day == 1`. Do you think that's confusing?

I believe it's adequately covered by test_fiscalyear:886:
`assert a.start == fiscalyear.FiscalYear(a.fiscal_year).start`

**Comment by nicmendoza** [comment]
Added some simple documentation.

Does it make sense to add sections for FiscalMonth and FiscalDay to the readme? Not sure why the thought didn't occur to me with the last PR, but even if it's a bit redundant it might help someone identify that the library is relevant for their use case.

Also, I added very subtle documentation for the `isleap` attribute I needed to add to FiscalYear. I'm not sure anyone will ever find it terribly useful, but it felt weird not documenting it so I sort of snuck it into the end of the example for FiscalYear.

File: README.rst [hunk 9]
```diff
@@ -80,13 +84,15 @@ You can also get the current ``FiscalQuarter`` with:
 
 .. code-block:: python
 
+   >>> from fiscalyear import *
```
Comment (adamjstewart) [inline README.rst:87]: I think you can remove this line, the README already assumes that was imported at the top.

File: fiscalyear.py [hunk 10]
```diff
@@ -674,6 +715,8 @@ def __contains__(self, item):
         """
         if isinstance(item, FiscalMonth):
             return self == item
+        if isinstance(item, FiscalDay):
```
Comment (adamjstewart) [inline fiscalyear.py:718]:
```suggestion
        elif isinstance(item, FiscalDay):
```

**Comment by adamjstewart** [comment]
> Does it make sense to add sections for FiscalMonth and FiscalDay to the readme?

Yes, please add to the README.rst and basics.rst, either in this PR or a follow-up PR.

> Also, I added very subtle documentation for the isleap attribute I needed to add to FiscalYear. I'm not sure anyone will ever find it terribly useful, but it felt weird not documenting it so I sort of snuck it into the end of the example for FiscalYear.

I think that's good. It will also appear in the Sphinx-generated API docs.

File: README.rst (hunk 9 above)
Comment (nicmendoza) [reply README.rst:87]: It came through when I copied and pasted the content from basic_usage.rst. Figured I'd leave it in and see what you felt. Happy to remove it though.

Comment (adamjstewart) [reply README.rst]: Can you make `basic_usage.rst` the same? We should only import things once.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**